karamel-core/src/main/java/se/kth/karamel/backend/github/Github.java

```diff
@@ -0,0 +1,476 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package se.kth.karamel.backend.github;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.commons.io.FileUtils;
+import org.eclipse.egit.github.core.Repository;
+import org.eclipse.egit.github.core.SearchRepository;
+import org.eclipse.egit.github.core.User;
+import org.eclipse.egit.github.core.client.GitHubClient;
+import org.eclipse.egit.github.core.service.OrganizationService;
+import org.eclipse.egit.github.core.service.RepositoryService;
+import org.eclipse.egit.github.core.service.UserService;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import org.kohsuke.github.GHOrganization;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import se.kth.karamel.common.Confs;
+import se.kth.karamel.common.CookbookScaffolder;
+import se.kth.karamel.common.Settings;
+import se.kth.karamel.common.exception.KaramelException;
+
+/**
+ * 1. Call registerCredentials() to store your github credentials in memory. 2. Then call methods like addFile(),
+ * commitPush(repo,..)
+ *
+ */
+public class Github {
```

Comment:
Since we have several Github related classes, a more specific naming as well as a sentence describing the main function of the class is appreciated.